src/kinetics/FalloffFactory.cpp

```diff
@@ -34,10 +37,23 @@ Falloff* FalloffFactory::newFalloff(int type, const vector_fp& c)
     return f;
 }
 
+Falloff* FalloffFactory::newFalloff(std::string type, const vector_fp& c)
+{
+    Falloff* f = create(type);
+    f->init(c);
+    return f;
+}
+
 shared_ptr<Falloff> newFalloff(int type, const vector_fp& c)
 {
     shared_ptr<Falloff> f(FalloffFactory::factory()->newFalloff(type, c));
     return f;
 }
 
+shared_ptr<Falloff> newFalloff(std::string type, const vector_fp& c)
```

Comment:
It should be possible to avoid an unnecessary copy with:
```suggestion
shared_ptr<Falloff> newFalloff(const std::string& type, const vector_fp& c)
```